src/components/VMapbox.vue

```diff
@@ -211,9 +211,10 @@ export default {
       this.addLayers()
     })
 
-    // If the style was changed, wait for the styledata to be loaded and re-add all the layers
+    // If the style was changed, wait for the style to be loaded and re-add all the layers
+    // https://github.com/mapbox/mapbox-gl-js/issues/4006
     this.$on('style:update', () => {
-      this.$once('mb-styledata', () => {
+      this.$once('mb-style.load', () => {
```

Comment:
On both the doc and the example they use `style.load` for this https://github.com/mapbox/mapbox-gl-js/blob/937f67f321f5419c0e82e78e262553330e59c8bd/src/ui/events.js#L1502